multinet/uploaders/csv.py

```diff
@@ -36,29 +43,34 @@ def validate_csv(rows: Sequence[MutableMapping]) -> None:
                 unique_keys.add(key)
 
         if len(duplicates) > 0:
-            data_errors.append({"error": "duplicate", "detail": list(duplicates)})
+            data_errors.append(
+                BasicError(type="csv_duplicate_keys", body=list(duplicates))
+            )
+
     elif "_from" in fieldnames and "_to" in fieldnames:
         # Edge Table, check that each cell has the correct format
         valid_cell = re.compile("[^/]+/[^/]+")
 
-        detail = []
+        invalid_syntax_errors: List[CSVInvalidRow] = []
 
         for i, row in enumerate(rows):
-            fields = []
+            fields: List[str] = []
             if not valid_cell.match(row["_from"]):
                 fields.append("_from")
             if not valid_cell.match(row["_to"]):
                 fields.append("_to")
 
             if fields:
                 # i+2 -> +1 for index offset, +1 due to header row
-                detail.append({"fields": fields, "row": i + 2})
+                invalid_syntax_errors.append({"fields": fields, "row": i + 2})
```

Comment:
Do you need to instantiate this as a new CSVInvalidRow or is it automatically converted from a dict to that class?

Reply:
It's technically not required because the type matches that of `CSVInvalidRow` (i.e., instantiating that produces the same dictionary), but we should for consistency. I'll change that